File: types/user.type.go
package types

// Types - data structures for user
// These structs represent requests, responses, and entities
// that are only meaningful within this api.

type User struct {
    ID ID                       `json:"id"`
    DisplayName string          `json:"display_name"`
    Email string                `json:"email"`
    Password string             `json:"password"`
    Avatar string               `json:"avatar"`
    SystemRole string           `json:"system_role"`
    Preference map[string]any   `json:"preference"`
    IsActive bool               `json:"is_active"`
    UpdatedAt string            `json:"updated_at"`
    CreatedAt string            `json:"created_at"`
}



type NewOwnerPayload struct {
    DisplayName string          `json:"display_name"`
    Email string                `json:"email"`
    Password string             `json:"password"`
}


type UpdateAccountNamePayload struct {
    DisplayName string          `json:"display_name"`
}
type UpdateAccountPasswordPayload struct {
    Password string             `json:"password"`
    NewPassword string          `json:"new_password"`
}

File: ui/src/types/user.type.ts
import type { SystemRolesType } from "@/constants";


export interface UserModel {
    id: string;
    display_name: string;
    email: string;
    avatar: string;
    system_role: SystemRolesType;
    preference: Object;
    is_active: boolean;
    updated_at: Date;
    created_at: Date;
}

export interface NewOwnerPayload {
    display_name: string;
    email: string;
    password: string;
    confirm_password: string;
}


export interface NewUserPayload {
    display_name: string;
    email: string;
    password: string;
    system_role: SystemRolesType;
    avatar: string;
}


export interface UpdateUserPayload {
    display_name: string;
    email: string;
    system_role: SystemRolesType
}